### PR TITLE
browser: notify webcontents hidden when window not shown

### LINKED ISSUE #2826: `document.hidden` is incorrect if window created but not shown
Currently, `document.hidden` is set as `false` if a new `BrowserWindow` is created with `show: false`.
The value is set properly once the window is shown and subsequently hidden again.

#### Repro
``` js
var app = require('app');
var Window = require('browser-window');

app.on('ready', function () {
  var window = new Window({ width: 200, height: 200, show: false });
  var page = window.webContents;

  page.on('did-finish-load', function () {
    page.executeJavaScript(`
      console.log('hidden: ' + document.hidden);
      require('remote').process.exit();
    `);
  });

  window.loadUrl('about:blank');
});
```

``` sh
$ electron testcase.js
[76450:0918/002034:INFO:renderer_main.cc(200)] Renderer process started
[76448:0918/002034:INFO:CONSOLE(2)] "hidden: false", source:  (2)
```

#### Workaround
``` js
var win = require('remote').getCurrentWindow();
Object.defineProperty(document, 'hidden', {
  get() {
    return !(win.isVisible());
  }
});
```

### PR DESCRIPTION
**Pull request by deepak1556**
Fixes #2826

### CODE REVIEW

File: atom/browser/native_window.cc [hunk 1]
```diff
@@ -159,8 +159,13 @@ void NativeWindow::InitFromOptions(const mate::Dictionary& options) {
   // Then show it.
   bool show = true;
   options.Get(switches::kShow, &show);
-  if (show)
+  if (show) {
     Show();
+  } else {
+    // When RenderView is created it sets to visible, this is to prevent
+    // breaiking the visibility API.
```
Comment (hokein) [inline atom/browser/native_window.cc]: A typo: `breaking`

Comment (deepak1556) [inline atom/browser/native_window.cc]: FIxed, thanks!

File: spec/chromium-spec.coffee [hunk 2]
```diff
@@ -23,6 +23,12 @@ describe 'chromium feature', ->
         {port} = server.address()
         $.get "http://127.0.0.1:#{port}"
 
+  describe 'document.hidden', ->
+    it 'is set correctly when window is not shown', ->
+      CI = remote.process.argv[2]
+      return if CI != '--ci'
```
Comment (steelbrain) [inline spec/chromium-spec.coffee]: why not use coffeescript operator `isnt`?

**Comment by zcbenz** [comment]
Can you write a spec for this by creating a hidden BrowserWindow, like the other BrowserWindow specs?

**Comment by deepak1556** [comment]
@zcbenz fixed.

**Comment by zcbenz** [comment]
:+1:

**Pull request merged**